File: ComfyStudio_StudioOS_CURRENT/frontend/src/director_os/prompt_engine/compiler.ts


import {
PromptContext,
CompiledPrompt
}
from "./schema";



export function compilePrompt(
ctx:PromptContext
):CompiledPrompt{



let positive=`

${ctx.year}年代真人影视画面，

角色:
${ctx.character.join(",")}

场景:
${ctx.scene}

道具:
${ctx.props.join(",")}

情绪:
${ctx.emotion}

镜头:
${ctx.camera}

时长:
${ctx.duration}秒

`;




let negative=`

禁止年龄漂移，

禁止人物身份变化，

禁止现代物品进入旧年代，

禁止错误电话设备，

禁止空间穿越，

禁止多人融合。



`;



return {


positive,


negative,


references:
ctx.referencePack



};



}

File: ComfyStudio_StudioOS_CURRENT/backend/modules/prompt/compiler.py
class PromptCompiler:


    def compile(shot):


        positive=f"""

电影级写实短剧镜头

人物:
{shot.get("character","")}

场景:
{shot.get("scene","")}

动作:
{shot.get("action","")}

比例:
9:16

24fps

"""

        negative="""

年龄错误

年代错误

现代设备

人物漂移

服装错误

"""

        return {

        "positive":positive,

        "negative":negative

        }

